azure/services/async/async.go

```diff
@@ -85,7 +94,7 @@ func processOngoingOperation(ctx context.Context, scope FutureScope, client Futu
 		// This can happen if the resource was deleted by another process before we could get the result.
 		scope.DeleteLongRunningOperationState(resourceName, serviceName)
 	}
-	return result, err
+	return result, resultErr{err}
```

Comment:
instead of having to differentiate the error here, it might be easier to call `DeleteLongRunningOperationState` in all cases where done is true (regardless of the error return) that way we'll attempt a new operation on the next loop if the current one finished and won't keep the erroneous future in Status.